### PR TITLE
CLOUD-2459 - correct documentation links and provide README information

### PR DESCRIPTION
**Pull request by luck3y**
Signed-off-by: Ken Wills kwills@redhat.com

https://issues.jboss.org/browse/CLOUD-2459

### CODE REVIEW

**Review by bstansberry — APPROVED** [review]
That's a good improvement.

**Comment by luck3y** [comment]
@bstansberry sorry for the second push. While looking at https://github.com/openshift/library/blob/master/official.yaml I noticed their use of the autogenerated template docs, so this adds those as well.

(the py script generates these, its a modified version taken from application_templates).

**Review by bstansberry — APPROVED** [review]

File: template.adoc.in [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+|============
+{/replicas}
+
+==== Pod Template
+
+{#serviceAccountName}
+===== Service Accounts
+
+Service accounts are API objects that exist within each project. They can be created or deleted like any other API object. Refer to the
+https://docs.openshift.com/enterprise/3.0/dev_guide/service_accounts.html#managing-service-accounts[Openshift documentation] for more
```
Comment (maschmid) [inline template.adoc.in]: we should link to a more recent version of the openshift docs

File: template.adoc.in [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+{/volumes}
+
+=== External Dependencies
+
+{#PersistentVolumeClaim}
+==== Volume Claims
+
+A `PersistentVolume` object is a storage resource in an OpenShift cluster. Storage is provisioned by an administrator
+by creating `PersistentVolume` objects from sources such as GCE Persistent Disks, AWS Elastic Block Stores (EBS), and NFS mounts.
+Refer to the https://docs.openshift.com/enterprise/3.0/dev_guide/persistent_volumes.html#overview[Openshift documentation] for
```
Comment (maschmid) [inline template.adoc.in]: also here, lets use the current OCP version link instead

File: README.adoc [hunk 3]
```diff
@@ -2,7 +2,55 @@
...
+## Creating Secrets
+
+Some of the included templates require the creation of secrets.
+
+Example:
+[source, bash]
+----
+$ keytool -genkey -keyalg RSA -alias eapdemo-selfsigned -keystore keystore.jks -validity 360 -keysize 2048
+$ oc secrets new eap7-app-secret keystore.jks
+$ oc secrets link eap-service-account eap7-app-secret
```
Comment (maschmid) [inline README.adoc]: we use the default service account now in the templates.

also the linking isn't required AFAIK

File: template.adoc.in [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+|Name | Access Mode
+{table}
+|=============
+
+{/PersistentVolumeClaim}
+
+{#secrets}
+==== Secrets
+
+This template requires link:../secrets/{secretName}.adoc[{secretFile}]
```
Comment (maschmid) [inline template.adoc.in]: this doesn't exist in our repo

**Comment by luck3y** [comment]
@maschmid corrected and regenerated docs, thanks

**Review by rcernich — APPROVED** [review]
Looks okay to me

**Pull request merged**